### ISSUE TITLE
Unpackfs fails with squashfs file, but works if especifying loop device

### ISSUE THREAD

**Issue by lluisgras** [calamares/calamares#1249]
Unpackfs fails with squashfs file, but works if especifying loop device
root@debian:~# cat /etc/calamares/modules/unpackfs.conf 
---
unpack:
    -   source: "/mnt/usb1/Debian-22092019.squashfs"
        sourcefs: "squashfs"
        destination: ""
        
root@debian:~# ls /mnt/usb1/
boot  Debian-22092019.squashfs	initrd	vmlinuz

root@debian:~# cat .cache/calamares/session.log (partial log)
.
.
.

2019-09-23 - 22:28:05 [8]: [JOBQUEUE]: Progress Overall:  19.3548 % (accumulated) +  0 % (this job) =  19.3548 % (total) 
2019-09-23 - 22:28:05 [6]: Python Error:
 <class 'subprocess.CalledProcessError'> 
 None 
 File "/usr/lib/x86_64-linux-gnu/calamares/modules/unpackfs/main.py", line 360, in run
    return unpackop.run()

  File "/usr/lib/x86_64-linux-gnu/calamares/modules/unpackfs/main.py", line 213, in run
    self.mount_image(entry, imgmountdir)

  File "/usr/lib/x86_64-linux-gnu/calamares/modules/unpackfs/main.py", line 262, in mount_image
    "-o", "loop"

  File "/usr/lib/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd) 

.
.
.
    
Trying "-o", "loop=/dev/loop1,ro" in /usr/lib/x86_64-linux-gnu/calamares/modules/unpackfs/main.py, line 262 doesn't help.

With loop device /dev/loop1

root@debian:~# cat /etc/calamares/modules/unpackfs.conf
---
unpack:
    -   source: "/dev/loop1"
        sourcefs: "squashfs"
        destination: ""

root@debian:~# losetup -a
/dev/loop0: [2067]:18 (/mnt/dispositiu/Debian-22092019.squashfs) 

root@debian:~# losetup -f
/dev/loop1

root@debian:~# losetup /dev/loop1 /mnt/usb1/Debian-22092019.squashfs 

root@debian:~# losetup -a
/dev/loop1: [2067]:18 (/mnt/usb1/Debian-22092019.squashfs)
/dev/loop0: [2067]:18 (/mnt/dispositiu/Debian-22092019.squashfs)

Now, works like a charm

root@debian:~# cat .cache/calamares/session.log (partial log)
.
.
.

2019-09-23 - 22:45:47 [6]: Starting job "umount" 
2019-09-23 - 22:45:47 [6]: Determining gettext path from ("/root", "/usr/bin", "/usr/share/locale") 
2019-09-23 - 22:45:47 [6]:  ..  Found "ca_ES" in "/usr/share/locale/ca_ES" 
2019-09-23 - 22:45:47 [6]: Job file "/usr/lib/x86_64-linux-gnu/calamares/modules/umount/main.py" 
2019-09-23 - 22:45:47 [6]: Job description from pretty_name "umount" = "Desmunta els sistemes de fitxers." 
2019-09-23 - 22:45:47 [8]: [JOBQUEUE]: Progress for Job[ 30 ]:  0 % completed 
2019-09-23 - 22:45:47 [8]: [JOBQUEUE]: Progress Overall:  96.7742 % (accumulated) +  0 % (this job) =  96.7742 % (total) 
2019-09-23 - 22:45:48 [6]: FinishedPage::setUpRestart(), Quit button setup= "user-checked" command= "systemctl -i reboot" 
2019-09-23 - 22:46:00 [8]: Shutting down Calamares... 
2019-09-23 - 22:46:00 [8]:  ..  Finished shutdown.

**Comment by adriaandegroot** [comment]
I really can't tell what you're trying to say here. The failing code is:

```
            subprocess.check_call(["mount",
                                   entry.source,
                                   imgmountdir,
                                   "-t", entry.sourcefs,
                                   "-o", "loop"
                                   ])
```

Which .. well, *imgmountdir* is the temp place where Calamares puts these images. You could `mkdir /tmp/experiment` and use that. Can you mount your squashfs then?

**Comment by lluisgras** [comment]
My Linux distro is based on Debian Buster and during boot, it puts the USB content in /mnt/usb1/
The first tests gave me an error if I indicated in /etc/calamares/modules/unpackfs.conf
---
unpack:
    -   source: "/mnt/usb1/BOX-08102019.squashfs"
        sourcefs: "squashfs"
        destination: ""

But they worked if it indicated the loop device associated with the squashfs file with losetup.

---
unpack:
    -   source: "/dev/loop1"
        sourcefs: "squashfs"
        destination: ""

I just tried it with the debian 3.2.13-1 package and now it works correctly. Both using the squashfs file or the loop device associated with it. 

I do not know the reason for the previous error but it is replayed if I disconnect the loop device with losetup -D.

If i put /tmp/something in unpackfs.conf

destination: "/tmp/something"

Installer complaints about /tmp/calamares-?????/tmp/something doesn't exist.
